packages/flutter/test/widgets/layout_builder_test.dart

```diff
@@ -175,4 +293,300 @@ void main() {
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: CustomScrollView(
+          controller: scrollController,
+          slivers: <Widget>[
+            const SliverToBoxAdapter(
+              child: SizedBox(height: 300),
+            ),
+            SliverLayoutBuilder(
+              builder: (BuildContext context, SliverConstraints constraint) => SliverToBoxAdapter(
+                child: SizedBox(key: childKey1, height: 200),
+              ),
+            ),
+            SliverToBoxAdapter(
+              child: SizedBox(key: childKey2, height: 100),
+            ),
+          ],
+        ),
+      ),
+    );
+
+    final RenderBox renderChild1 = tester.renderObject(find.byKey(childKey1));
+    final RenderBox renderChild2 = tester.renderObject(find.byKey(childKey2));
+
+    // scrollController.scrollOffset = 0:
+    expect(
+      renderChild1.localToGlobal(const Offset(100, 100)),
+      const Offset(100, 300.0 + 100),
+    );
+
+    expect(
+      renderChild2.localToGlobal(const Offset(100, 100)),
+      const Offset(100, 300.0 + 200 + 100),
+    );
+
+    scrollController.jumpTo(100);
+    await tester.pump();
+    expect(
+      renderChild1.localToGlobal(const Offset(100, 100)),
+      // -100 because the scroll offset is now 100.
+      const Offset(100, 300.0 + 100 - 100),
+    );
+
+    expect(
+      renderChild2.localToGlobal(const Offset(100, 100)),
+      // -100 because the scroll offset is now 100.
+      const Offset(100, 300.0 + 100 + 200 - 100),
+    );
+  });
+
+  testWidgets('hitTest works within SliverLayoutBuilder', (WidgetTester tester) async {
+    final ScrollController scrollController = ScrollController();
+    List<int> hitCounts = <int> [0, 0, 0];
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Padding(
+          padding: const EdgeInsets.all(50),
+          child: CustomScrollView(
+            controller: scrollController,
+            slivers: <Widget>[
+              SliverToBoxAdapter(
+                child: SizedBox(
+                  height: 200,
+                  child: GestureDetector(onTap: () => hitCounts[0]++),
+                ),
+              ),
+              SliverLayoutBuilder(
+                builder: (BuildContext context, SliverConstraints constraint) => SliverToBoxAdapter(
+                  child: SizedBox(
+                    height: 200,
+                    child: GestureDetector(onTap: () => hitCounts[1]++),
+                  ),
+                ),
+              ),
+              SliverToBoxAdapter(
+                child: SizedBox(
+                  height: 200,
+                  child: GestureDetector(onTap: () => hitCounts[2]++),
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+
+    // Tap item 1.
+    await tester.tapAt(const Offset(300, 50.0 + 100));
```

Comment:
lets use tap(find.byType(SizedBox).at(index))

Reply:
same for all other tapAt

Reply:
The point of using `tapAt` here is to verify the hit test results of the exact `Offset's are correct, tapping at the center of a `RenderBox` may not be good enough. Moreover some of these `RenderBox`es are only partially visible, so their centers may not be in the viewport, I don't think `tap` works in that case?

Reply:
I see, sounds good to me